modules/apps/portal-security-sso/portal-security-sso-openid-connect-impl/src/main/java/com/liferay/portal/security/sso/openid/connect/internal/configuration/admin/service/OpenIdConnectProviderManagedServiceFactory.java

```diff
@@ -475,31 +506,37 @@ private void _updateOAuthClientEntry(
 		}
 
 		try {
-			String authServerWellKnownURI = _updateOAuthClientASLocalMetadata(
-				defaultUserId, properties);
-
-			OAuthClientEntry oAuthClientEntry =
-				_oAuthClientEntryLocalService.fetchOAuthClientEntry(
-					companyId, authServerWellKnownURI,
-					_getPropertyAsString("openIdConnectClientId", properties));
-
-			if (oAuthClientEntry == null) {
-				oAuthClientEntry =
-					_oAuthClientEntryLocalService.addOAuthClientEntry(
-						defaultUserId,
-						_generateAuthRequestParametersJSON(properties),
-						authServerWellKnownURI, _generateInfoJSON(properties),
-						OAuthClientEntryConstants.OIDC_USER_INFO_MAPPER_JSON,
-						_generateTokenRequestParametersJSON(properties));
+			OAuthClientEntry oAuthClientEntry = null;
+
+			if (oldProperties != null) {
+				String oldAuthServerWellKnownURI =
+					_updateOAuthClientASLocalMetadata(
+						defaultUserId, properties);
+
+				OAuthClientEntry oldOAuthClientEntry =
+					_oAuthClientEntryLocalService.fetchOAuthClientEntry(
+						companyId, oldAuthServerWellKnownURI,
+						_getPropertyAsString(
+							"openIdConnectClientId", oldProperties));
+
+				if (oldOAuthClientEntry != null) {
+					oAuthClientEntry =
+						_oAuthClientEntryLocalService.updateOAuthClientEntry(
+							oldOAuthClientEntry.getOAuthClientEntryId(),
+							_generateAuthRequestParametersJSON(properties),
+							oldAuthServerWellKnownURI,
+							_generateInfoJSON(properties),
+							oldOAuthClientEntry.getOIDCUserInfoMapperJSON(),
+							_generateTokenRequestParametersJSON(properties));
+				}
+				else {
+					oAuthClientEntry = _addOrUpdateOAuthClientEntry(
+						companyId, properties, defaultUserId);
```

Comment:
Do you know if it's a valid case that we don't have `oldProperties` but we need to update an object? I'm not sure but if it's impossible than we can eliminate `_addOrUpdateOAuthClientEntry` and use only the add parts instead.